src/hotspot/cpu/x86/gc/shared/barrierSetAssembler_x86.cpp

```diff
@@ -195,6 +195,99 @@ void BarrierSetAssembler::store_at(MacroAssembler* masm, DecoratorSet decorators
   }
 }
 
+void BarrierSetAssembler::copy_load_at(MacroAssembler* masm,
+                                       DecoratorSet decorators,
+                                       BasicType type,
+                                       size_t bytes,
+                                       Register dst,
+                                       Address src,
+                                       Register tmp) {
+  assert(bytes <= 8, "can only deal with non-vector registers");
+  if (bytes == 1) {
+    __ movb(dst, src);
+  } else if (bytes == 2) {
+    __ movw(dst, src);
+  } else if (bytes == 4) {
+    __ movl(dst, src);
+  } else if (bytes == 8) {
+#ifdef _LP64
+    __ movq(dst, src);
+#else
+    fatal("No support for 8 bytes copy");
+#endif
+  }
```

Comment:
I believe the intent would be clearer using `switch`.

Reply:
I tried both, but I actually found it less clear with switch as I then have to worry about a missing break; which would be accepted by the compiler and just yield subtly incorrect machine code. I'm open to doing switch though if you really do think it is more clear then. What do you think?

Reply:
I think `switch` looks better; it's not that this code will be updated frequently, adding/removing cases. (Ofc, this is subjective.)

```cpp
switch(byte) {
case 1: ...; break;
case 2: ...; break;
...
default: should-not-reach;
}